### PR TITLE
Feat/toc update

### PR DESCRIPTION
**Pull request by RaphaelAmadeu**
### Related Ticket(s)

#1851 

### Description

I've updated `TableOfContents` so it reflects the new design and functional specs.
The optional children and rule were added.
G10 theme was added as well.

### Changelog

**New**

- Optional children and rule

### CODE REVIEW

File: packages/react/src/patterns/sub-patterns/TableOfContents/TOCDesktop.js [hunk 1]
```diff
@@ -85,6 +87,18 @@ const TOCDesktop = ({ menuItems, selectedId }) => {
     <div
       className={`${prefix}--tableofcontents__desktop`}
       data-autoid={`${stablePrefix}--tableofcontents__desktop`}>
+      {children ? (
+        <div className={`${prefix}--tableofcontents__desktop__children`}>
+          {children}
+        </div>
+      ) : (
+        false
+      )}
```
Comment (asudoh) [inline packages/react/src/patterns/sub-patterns/TableOfContents/TOCDesktop.js:96]:
```suggestion
      ) : null}
```

File: packages/react/src/patterns/sub-patterns/TableOfContents/TOCDesktop.js [hunk 2]
```diff
@@ -85,6 +87,18 @@ const TOCDesktop = ({ menuItems, selectedId }) => {
...
+      {menuRule ? (
+        <hr className={`${prefix}--tableofcontents__desktop__rule`} />
+      ) : (
+        false
+      )}
```
Comment (asudoh) [inline packages/react/src/patterns/sub-patterns/TableOfContents/TOCDesktop.js:101]:
```suggestion
      ) : null}
```

File: packages/react/src/patterns/sub-patterns/TableOfContents/TableOfContents.js [hunk 3]
```diff
@@ -202,6 +205,7 @@ const TableOfContents = ({
     selectedTitle,
     menuLabel,
     updateState,
+    children: children.length > 1 ? children[0] : false,
```
Comment (asudoh) [inline packages/react/src/patterns/sub-patterns/TableOfContents/TableOfContents.js:208]:
```suggestion
    children: children.length > 1 ? children[0] : null,
```

File: packages/react/src/patterns/sub-patterns/TableOfContents/README.md [hunk 4]
```diff
@@ -83,11 +83,26 @@ Add the following line in your `.env` file at the root of your project.
 
 | Name           | Required | Data Type | Default Value | Description                                                                      |
 | -------------- | -------- | --------- | ------------- | -------------------------------------------------------------------------------- |
-| `children`     | NO       | Array     | null          | Content to display next to the side nav.                                         |
+| `children`     | NO       | Array     | null          | Content to display next to the side nav. (See `Children` bellow)                 |
```
Comment (jeffchew) [inline packages/react/src/patterns/sub-patterns/TableOfContents/README.md:86]: I wonder if we should call this something else other than `children`. Children is also a reserved name for the actual children content that goes into the content area of the Table of Contents. 

Maybe we should rename to something like `headingContent`?

File: packages/react/src/patterns/sub-patterns/TableOfContents/README.md [hunk 5]
```diff
@@ -83,11 +83,26 @@ Add the following line in your `.env` file at the root of your project.
...
 | `menuItems`    | NO       | Array     | null          | Array of menu item objects to render within the side nav. See `menuItems` below. |
 | `menuLabel`    | NO       | String    | `Jump to`     | Placeholder value for menu label                                                 |
 | `theme`        | NO       | String    | `white`       | Defines the color theme for the pattern (See `Theme` bellow)                     |
 | `stickyOffset` | NO       | Number    | null          | Defines the offset for the sticky column                                         |
+| `menurule`     | NO       | Boolean   | null          | Defines if the menu ruler will be rendered                                       |
+
+##Children
+
+`TableOfContents` supports an aditional `children` prop, that is rendered above
```
Comment (jeffchew) [inline packages/react/src/patterns/sub-patterns/TableOfContents/README.md:95]: `aditional` misspelled

File: packages/react/src/patterns/sub-patterns/TableOfContents/__stories__/TableOfContents.stories.js [hunk 6]
```diff
@@ -57,7 +63,22 @@ export const ManuallyDefineMenuItems = () => {
     <TableOfContents
       theme={select('theme', _themes, _themes.white)}
       menuItems={object('menuItems', menuItems)}
-      menuLabel={_menuLabel}>
+      menuLabel={_menuLabel}
+      menuRule={boolean('Optional Rule', false)}>
+      {boolean('Optional children', false) ? (
```
Comment (jeffchew) [inline packages/react/src/patterns/sub-patterns/TableOfContents/__stories__/TableOfContents.stories.js:68]: @RaphaelAmadeu let's actually create this as a separate story, so it's captured in a separate percy snapshot.

File: packages/react/src/patterns/sub-patterns/TableOfContents/README.md [hunk 7]
```diff
@@ -81,13 +86,15 @@ Add the following line in your `.env` file at the root of your project.
...
+| ---------------- | -------- | --------- | ------------- | -------------------------------------------------------------------------------- |
+| `children`       | NO       | Array     | null          | Content to display next to the side nav.                                         |
+| `headingContent` | NO       | Component | null          | Content to be displayed above the navigation menu.                               |
+| `menuItems`      | NO       | Array     | null          | Array of menu item objects to render within the side nav. See `menuItems` below. |
+| `menuLabel`      | NO       | String    | `Jump to`     | Placeholder value for menu label                                                 |
+| `theme`          | NO       | String    | `white`       | Defines the color theme for the pattern (See `Theme` bellow)                     |
+| `stickyOffset`   | NO       | Number    | null          | Defines the offset for the sticky column                                         |
+| `menurule`       | NO       | Boolean   | null          | Defines if the menu ruler will be rendered                                       |
```
Comment (jeffchew) [inline packages/react/src/patterns/sub-patterns/TableOfContents/README.md:97]: Super minor, it looks like it should be `menuRule`

Comment (RaphaelAmadeu) [reply packages/react/src/patterns/sub-patterns/TableOfContents/README.md]: @jeffchew I've corrected it, thank you!

File: packages/react/src/patterns/sub-patterns/TableOfContents/__stories__/TableOfContents.stories.js [hunk 8]
```diff
@@ -78,3 +99,54 @@ export const DynamicItems = () => {
...
+      <h4>Example children</h4>
+    </div>
+  );
+
+  return (
+    <TableOfContents
+      theme={select('theme', _themes, _themes.white)}
+      menuItems={object('menuItems', menuItems)}
+      menuLabel={_menuLabel}
+      menuRule={boolean('Optional Rule', false)}
```
Comment (jeffchew) [inline packages/react/src/patterns/sub-patterns/TableOfContents/__stories__/TableOfContents.stories.js:147]:
```suggestion
      menuRule={boolean('Optional Rule (menuRule)', false)}
```

File: packages/react/src/patterns/sub-patterns/TableOfContents/__stories__/TableOfContents.stories.js [hunk 9]
```diff
@@ -78,3 +99,54 @@ export const DynamicItems = () => {
     </TableOfContents>
   );
 };
+
+export const WithHeadingContent = () => {
+  const _menuLabel = text('menu label', 'Jump to');
```
Comment (jeffchew) [inline packages/react/src/patterns/sub-patterns/TableOfContents/__stories__/TableOfContents.stories.js:104]:
```suggestion
  const _menuLabel = text('menu label (menuLabel)', 'Jump to');
```

File: packages/react/src/patterns/sub-patterns/TableOfContents/TOCDesktop.js [hunk 10]
```diff
@@ -21,9 +21,10 @@ const { prefix } = settings;
  * @param {object} props props object
  * @param {Array} props.menuItems menu items object
  * @param {string} props.selectedId id of a menu item
+ * @param {boolean} props.menuRule optional rule
```
Comment (jeffchew) [inline packages/react/src/patterns/sub-patterns/TableOfContents/TOCDesktop.js:24]: Can you add headingContent to the JSDocs?

**Comment by jeffchew** [comment]
Screen Shot 2020-05-21 at 5 52 51 PM

Is it intentional to have the children in the right side? Also, we probably don't need the knobs on the manually defined menu item story if we have a separate story for the heading content.

File: packages/react/src/patterns/sub-patterns/TableOfContents/__tests__/TableOfContents.test.js [hunk 11]
```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import dataContent from '../../../../../es/patterns/sub-patterns/TableOfContents/__stories__/data/dataContent';
```
Comment (jeffchew) [inline packages/react/src/patterns/sub-patterns/TableOfContents/__tests__/TableOfContents.test.js:8]: @RaphaelAmadeu for the data content, you don't have to reference the es folder.

**Pull request merged**